### PR TITLE
engineering: Validation with Zod

### PR DESCRIPTION
**Pull request by namtrhg**
#### What's this PR do?

- [x] Explain Zod validation library
- [x] Explain why people use Zod
- [x] Explain how to use Zod
- [x] Compare Zod with Yup

### CODE REVIEW

**Comment by tienan92it** [comment]
LGTM

**Review by tienan92it — APPROVED** [review]

File: Engineering/Frontend/Validation with Zod.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
+---
+tags: engineering, frontend, validation, parsing, typescript, zod, yup
+author: Tran Hoang Nam
+github_id: namtrhg
+date: 2022-01-12
```
Comment (tienan92it) [inline Engineering/Frontend/Validation with Zod.md:5]: is it 2023?

Comment (namtrhg) [reply Engineering/Frontend/Validation with Zod.md:5]: yes

Comment (namtrhg) [reply Engineering/Frontend/Validation with Zod.md:5]: will fix it

**Review by janrvu — COMMENTED** [review]
The note is presented in the correct format.

**Review by monotykamary — APPROVED** [review]
LGTM

**Pull request merged**